atomic_reactor/plugins/pre_add_flatpak_labels.py

```diff
@@ -0,0 +1,55 @@
+"""
+Copyright (c) 2020 Red Hat, Inc
+All rights reserved.
+
+This software may be modified and distributed under the terms
+of the BSD license. See the LICENSE file for details.
+
+
+Pre build plugin which adds additional labels to the Dockerfile automatically
+created for a flatpak, based on the flatpak: labels key in container.yaml.
+"""
+
+from __future__ import unicode_literals, absolute_import
+
+from atomic_reactor.plugin import PreBuildPlugin
+from atomic_reactor.util import df_parser, label_to_string
+
+
+class AddFlatpakLabelsPlugin(PreBuildPlugin):
+    key = "add_flatpak_labels"
+    is_allowed_to_fail = False
+
+    def __init__(self, tasker, workflow):
+        """
+        constructor
+
+        :param tasker: ContainerTasker instance
+        :param workflow: DockerBuildWorkflow instance
+        """
+        # call parent constructor
+        super(AddFlatpakLabelsPlugin, self).__init__(tasker, workflow)
+
+    def run(self):
+        """
+        run the plugin
+        """
+        flatpak_yaml = self.workflow.source.config.flatpak
+        if flatpak_yaml is None:
+            return
+        labels = flatpak_yaml.get('labels')
+        if not labels:
+            return
+
+        dockerfile = df_parser(self.workflow.builder.df_path, workflow=self.workflow)
```

Comment:
It seems weird to keep recreating the df_parser for all these plugins that need to manipulate the Dockerfile, and we should probably at some point just store the df_parser in tasker or workflow.

but that's beyond the scope of this PR, I just wanted to make that observation.